debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsSnapshotProducerIT.java

```diff
@@ -480,6 +487,118 @@ public void shouldNotSnapshotMaterializedViews() throws Exception {
+    }
+
+    @Test(expected = ComparisonFailure.class)
+    @FixFor("DBZ-1413")
+    public void shouldNotSnapshotNestedDomainAliasTypeModifiersNotPropagated() throws Exception {
+        // The pgjdbc driver does not currently provide support for type modifier resolution
+        // when a domain type extends an existing domain type that extends a base type using
+        // explicit type modifiers.
+        TestHelper.execute("CREATE DOMAIN varbit2 AS varbit(3);");
+        TestHelper.execute("CREATE DOMAIN varbit2b AS varbit2;");
+        TestHelper.execute("CREATE TABLE alias_table (pk SERIAL, value varbit2b NOT NULL, PRIMARY KEY (pk));");
+        TestHelper.execute("INSERT INTO alias_table (value) values (B'101');");
+
+        buildNoStreamProducer(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.DECIMAL_HANDLING_MODE, DecimalHandlingMode.DOUBLE)
+                .with(PostgresConnectorConfig.INCLUDE_UNKNOWN_DATATYPES, true));
+
+        final TestConsumer consumer = testConsumer(1, "public");
+        consumer.await(TestHelper.waitTimeForRecords() * 30, TimeUnit.SECONDS);
+
+        // We would normally expect that the value column was created with a length of 3.
+        // However due to how the resolution works in the driver, it returns 2147483647.
+        // We probably want to avoid supporting this behavior for now?
+        List<SchemaAndValueField> expected = Collections.singletonList(
+                new SchemaAndValueField("value", Bits.builder(3).build(), new byte[]{ 5, 0 }));
+
+        consumer.process(record -> assertReadRecord(record, Collect.hashMapOf("public.alias_table", expected)));
+    }
+
+    @Test
+    @FixFor("DBZ-920")
+    public void shouldSnapshotEnumAsKnownType() throws Exception {
+        TestHelper.execute("CREATE TYPE test_type AS ENUM ('V1', 'V2');");
+        TestHelper.execute("CREATE TABLE enum_table (pk SERIAL, value test_type NOT NULL, primary key(pk));");
+        TestHelper.execute("INSERT INTO enum_table (value) values ('V1');");
+
+        // Specifically enable `column.propagate.source.type` here to validate later that the actual
+        // type, length, and scale values are resolved correctly when paired with Enum types.
+        buildNoStreamProducer(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.INCLUDE_UNKNOWN_DATATYPES, true)
+                .with(PostgresConnectorConfig.TABLE_WHITELIST, "public.enum_table")
+                .with("column.propagate.source.type", "public.enum_table.value"));
+
+        final TestConsumer consumer = testConsumer(1, "public");
+        consumer.await(TestHelper.waitTimeForRecords() * 30, TimeUnit.SECONDS);
+
+        List<SchemaAndValueField> expected = Collections.singletonList(
+                new SchemaAndValueField("value", Enum.builder("V1,V2")
+                        .parameter(TestHelper.TYPE_NAME_PARAMETER_KEY, "TEST_TYPE")
```

Comment:
Ah, so unlike as for domain types, for enum types the custom name is propagated?

Reply:
It actually propagates for both domain and enum types.  I'll update the test suite accordingly to account for this.